### PR TITLE
93 create all actions function on inputs

### PR DESCRIPTION
**Pull request by TomDUVAL-MAHE**
Implement actions functions on inputs. They will update some entities components. It will implement #93

Add Functions for : 
 - Shoot
 - MoovePlayerX
 - MoovePlayerY

### CODE REVIEW

**Review by LucasTesnier — APPROVED** [review]

File: libs/SFMLSystems/InputManagement.hpp [hunk 1]
```diff
@@ -18,7 +21,49 @@ namespace ecs
         /// @brief The run function of this system. It will be used by the world in the function runSystems.
         /// @param world The corresponding world on which run this system.
         void run(World &world) override final;
+
+        /// @brief This function is call when a player move his Abscissa with a input
+        /// @param world The world in which the Player is
+        /// @param move Action velocity 
+        inline void movePlayerX(World &world, float move)
```
Comment (31Nathan) [inline libs/SFMLSystems/InputManagement.hpp:28]: INLINE ?

File: libs/SFMLSystems/InputManagement.hpp [hunk 2]
```diff
@@ -18,7 +21,49 @@ namespace ecs
...
+            auto moveX = [move_d](std::shared_ptr<ecs::Entity> entityPtr) {
+                entityPtr->getComponent<Velocity>().multiplierAbscissa = move_d;
+            };
+            std::for_each(player.begin(), player.end(), moveX);
+        }
+
+        /// @brief This function is call when a player move his Ordinate with a input
+        /// @param world The world in which the Player is
+        /// @param move Action velocity 
+        inline void movePlayerY(World &world, float move)
```
Comment (31Nathan) [inline libs/SFMLSystems/InputManagement.hpp:42]: Crazy guy

File: libs/SFMLSystems/InputManagement.hpp [hunk 3]
```diff
@@ -18,7 +21,49 @@ namespace ecs
...
+            auto moveY = [move_d](std::shared_ptr<ecs::Entity> entityPtr) {
+                entityPtr->getComponent<Velocity>().multiplierOrdinate = move_d;
+            };
+            std::for_each(player.begin(), player.end(), moveY);
+        }
+
+        /// @brief This function is call when a player shoot with a input. It creates a Ally Projectile
+        /// @param world The world in which the Player is
+        /// @param move Action Shoot 
+        inline void shootAction(World &world, float action)
```
Comment (31Nathan) [inline libs/SFMLSystems/InputManagement.hpp:56]: I'm scared

**Review by 31Nathan — CHANGES_REQUESTED** [review]

File: libs/GameEntityManipulation/CreateEntitiesFunctions/CreatePlayer.hpp [hunk 4]
```diff
@@ -55,7 +56,8 @@ namespace ecs
                              .addComponent<DamageRadius>(damageRadius)
                              .addComponent<Collidable>()
                              .addComponent<Velocity>(multiplierAbscissa, multiplierOrdinate)
-                             .addComponent<Player>();
+                             .addComponent<Player>()
+                            .addComponent<Controlable>();
```
Comment (osvegn) [inline libs/GameEntityManipulation/CreateEntitiesFunctions/CreatePlayer.hpp:60]: missing space

**Review by osvegn — CHANGES_REQUESTED** [review]

**Review by LucasTesnier — APPROVED** [review]

**Review by 31Nathan — APPROVED** [review]
NTR

**Review by osvegn — APPROVED** [review]

**Pull request merged**